code/modules/mob/living/simple_animal/hostile/giant_spider/spiderling.dm

```diff
@@ -153,8 +107,8 @@
 		visible_message("<span class='danger'>The [src.name] tries to flee from [target.name]!</span>")
 
 /mob/living/simple_animal/hostile/giant_spider/spiderling/AttackingTarget()
-	if(istype(target, /obj/machinery/atmospherics/unary/vent_pump))
-		ventcrawl(target)
+	if(istype(target, /obj/machinery/atmospherics/unary/vent_pump) && !client)
```

Comment:
mind check like you originally wanted is better, but it's not a big deal here because the AI will never target vents for this mob